airflow-core/src/airflow/cli/commands/task_command.py

```diff
@@ -193,7 +194,10 @@ def _get_ti(
                 f"run_id or logical_date of {logical_date_or_run_id!r} not found"
             )
         # TODO: Validate map_index is in range?
-        ti = TaskInstance(task, run_id=dag_run.run_id, map_index=map_index)
+        dag_version = DagVersion.get_latest_version(dag.dag_id, session=session)
+        if TYPE_CHECKING:
+            assert dag_version
```

Comment:
That type checking assert feels really weird. Maybe just handle the None case `if ... is None: raise dag version not found, while we wait for the follow up PR to clean that up.